cfgov/data_research/forms.py

```diff
@@ -140,17 +142,38 @@ def save(self, commit=True):
             # Subscribe this registrant to GovDelivery.
             self.govdelivery_subscribe(code=self.govdelivery_code, email=email)
 
+            # Update their question response, if appropriate.
+            if self.govdelivery_question_id:
+                self.govdelivery_question_response(
```

Comment:
Unfortunately, because this requires two separate API calls, there's a possibility that an error on the second call would result in an incomplete registration.

Reply:
I did notice that, and I think it's probably desirable. If we're counting on the question response to indicate who's registered in GovDelivery, then a failure on the second call _is_ an incomplete registration.

Reply:
Right, I see. Such an error would leave someone in a state where they were signed up to the GD list, but not with the correct question ID, and also not recorded in the DB. We'd lose their registration information (company, etc). Hopefully they'd see the error message and re-submit, but if not, they'd end up being left on the mailing list. These kind of errors would (reasonably) require some manual review to check/fix.